python/truelayer_signing/sign.py

```diff
@@ -0,0 +1,82 @@
+from __future__ import annotations
+
+# std imports
+from dataclasses import dataclass
+from typing import Dict, Mapping, Optional
+
+# third party imports
+from jwt.algorithms import ECAlgorithm
+
+# local imports
+from .utils import HttpMethod, TlJwsBase, build_v2_jws_b64, to_url_safe_base64
+
+
+class TlSigner(TlJwsBase):
+    """
+    Tl-Signer
+    """
+    kid: str
+
+    def __init__(
+        self,
+        kid: str,
+        pkey: str,
+        method: HttpMethod = HttpMethod.POST,
+        path: str = "",
+        headers: Optional[Dict[str, str]] = None,
+        body: str = ""
+    ) -> None:
+        super().__init__(pkey, method, path, headers, body)
+        self.kid = kid
+
+    def sign(self) -> str:
+        """
+        Produce a JWS `Tl-Signature` v2.
+        """
+        return tl_sign(SignArguments(
+            self.kid,
+            self.pkey,
+            self.path,
+            self.headers,
+            self.body,
+            self.http_method
+        ))
+
+
+@dataclass(frozen=True)
+class SignArguments:
+    kid: str
+    pkey: str
+    path: str
+    headers: Mapping[str, str]
```

Comment:
Ordering is important here, since we're iterating over the keys to create `tl_headers` the user will need to use the same instance of headers when making the request (if using a `dict` as the `Mapping` instance) since the keys are not guaranteed to have the same ordering across different `dict` instances.

I think making this an `OrderedDict` would help making this explicit in the API though then users would be burdened with not being able to use a simple `dict`, alternatively we can add a notice in the docs/examples to make this very clear